codechecker_common/report.py

```diff
@@ -67,10 +68,34 @@ def __str__(self):
         msg += str(self.__files)
         return msg
 
-    def to_json(self):
+    def to_json(self, trim_path_prefixes):
         """ Converts to json format. """
         ret = self.__main
         ret["path"] = self.__bug_path
-        ret["files"] = self.__files
+
+        src_files = self.__files or []
+        ret["files"] = [util.trim_path_prefixes(file_path, trim_path_prefixes)
+                        for file_path in src_files]
 
         return ret
+
+    def to_codeclimate(self, trim_path_prefixes=None):
+        """ Convert to Code Climate format. """
+        location = self.main['location']
+        file_id = location['file']
+
+        file_path = self.files[file_id]
+        trimmed_source_file = \
+            util.trim_path_prefixes(file_path, trim_path_prefixes)
+
+        return {
```

Comment:
The codeclimate spec has some required fields
```
type -- Required. Must always be "issue".
check_name -- Required. A unique name representing the static analysis check that emitted this issue.
description -- Required. A string explaining the issue that was detected.
categories -- Required. At least one category indicating the nature of the issue being reported.
location -- Required. A Location object representing the place in the source code where the issue was discovered.
```
the fingerprint is required only by gitlab:
```
fingerprint -- Optional. A unique, deterministic identifier for the specific issue being reported to allow a user to exclude it from future analyses.
```
if the fields required by codeclimate spec are added can gitlab still parse the results?

I think adding the required fields would make the output a valid codeclimate format which the argument name suggests.

Reply:
Yes, GitLab will be able to parse the results with the extra fields (type, categories).